lib/aws/rails/sqs_active_job/executor.rb

```diff
@@ -36,8 +42,31 @@ def execute(message)
               job_msg = job ? "#{job.id}[#{job.class_name}]" : 'unknown job'
               @logger.info "Error processing job #{job_msg}: #{e}"
               @logger.debug e.backtrace.join("\n")
+            ensure
+              refresh_monitor = false
             end
           end
+
+          @monitor.post(message) do |message|
+            while refresh_monitor
+              begin
+                # Extend the visibility timeout to the provided refresh timeout
+                message.change_visibility({ visibility_timeout: @visibility_refresh })
+                # Wait half the refresh timeout, and repeat
+                sleep(@visibility_refresh / 2.0)
```

Comment:
also, does the `@monitor` thread need to have `shutdown` logic and be added to https://github.com/aws/aws-sdk-rails/blob/main/lib/aws/rails/sqs_active_job/poller.rb#L62-L64?

Reply:
Nah, as soon as the parent thread exits refresh_monitor will be set to false so this one will shut down. If a new job is picked up a new thread will be spun.

Reply:
> Nah, as soon as the parent thread exits refresh_monitor will be set to false so this one will shut down

but when will it shut down? if this thread is sleeping, it could be shut down way after the parent thread finishes